solutions/kattis-pointinpolygon.mdx

```diff
@@ -0,0 +1,151 @@
+
+</Info>
+
+We will define the determinant operator of two vectors as $\times$. Given two vectors $\vec{a}=\langle x_1,y_1 \rangle$ and $\vec{b}=\langle x_2,y_2 \rangle$. If $\vec{a}\times\vec{b}$ gives a positive area then $\vec{b}$ points to the left of $\vec{a}$, if it gives a negative area then $\vec{b}$ points to the right of $\vec{a}$, and if it gives a zero area then $\vec{b}$ and $\vec{a}$ are parallel.
+
+To check if $L_1$ and $L_2$ intersect with each other, is equal to checking if $L_2$ intersects with the line from line segment $L_1$, and checking if $L_1$ intersects with the line from line segment $L_2$. For the first of two condition to be true, $p_3$ and $p_4$ from $L_2$ must be on the opposite directions of $L_1$. To define this mathematically, let
+	
+$$
+\vec{a} = p_2-p_1
+$$
+$$
+\vec{b_1} = p_3-p_2
+$$
+$$
+\vec{b_2} = p_4-p_2
+$$
+
+The condition will be true if and only if $\vec{a}\times\vec{b_1} \neq \vec{a}\times\vec{b_2}$. Verifying the second of the two cases is equivalent to the procedure above, except $p_1$ and $p_2$ are switched with $p_3$ and $p_4$.
+
+As all geometry problem go, edge cases are a must. One edge case one might consider in our method above is that if both $\vec{a}\times\vec{b_1}$ and $\vec{a}\times\vec{b_2}$ are equal and are equal to 0, then they can still intersect as long as they overlap with each other. But let me in you a ***secret***, due to the way this problem is structured, all vertices are on lattice points, meaning that they have whole numbers as coordinates. And since one of the $L$ is customizable (we get to pick the point the point cast its ray at), we can just set its coordinate to $(\infty,1)$. This way, the ray will have a slope of close to $0$, successfully avoiding all possible lattice point intersections. One trick to rule edges cases all! 
+
+## Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+const int MAX = 1e5+5;
+
+struct Point{
+    ll x,y;
+};
+
+//Compute the determinant of two vectors (p1,p2) and (p3,p4)
+ll det(Point p1, Point p2, Point p3, Point p4){
+    return ((p2.x-p1.x)*(p4.y-p3.y)-(p2.y-p1.y)*(p4.x-p3.x));
+}
+
+//Check if point p3 is above or below line segment (p1,p2)
+ll dir(Point p1, Point p2, Point p3){
+    ll result = det(p1,p2,p2,p3);
+
+    if (result > 0){
+        return 1;
+    }else if (result < 0) {
+        return -1;
+    }else{
+        return 0;
+    }
+}
+
+//Check if two line segments (p1,p2) and (p3,p4) intersect
+bool intersect(Point p1, Point p2, Point p3, Point p4){
+    return ((dir(p1,p2,p3)!=dir(p1,p2,p4))&&(dir(p3,p4,p1)!=dir(p3,p4,p2)));
+}
+
+/*
+ * If point p3 is collinear with points p1 and p2
+ * Check if p3 is inside or outside the line segment (p1,p2)
+ */
+bool within(Point p1, Point p2, Point p3){
+    if (dir(p1,p2,p3)==0){
```

Comment:
tabs to indent, it's the convention around here
also use spaces around operators everywhere